pkg/controller/direct/dataflow/dataflowflextemplatejob_controller.go

```diff
@@ -330,30 +333,54 @@ func (a *dataflowFlexTemplateJobAdapter) Create(ctx context.Context, createOp *d
 	}
 
 	job := response.GetJob()
-	jobID := job.GetId()
-	// TODO: Use jobCreateTime for mutation checking?
-	// jobCreateTime := job.CreateTime()
 
-	ready := false
-	for !ready {
-		time.Sleep(2 * time.Second)
-		latest, err := a.getJob(ctx, jobID)
-		if err != nil {
-			// TODO: not right!
-			return fmt.Errorf("getting state of job")
+	if err := a.updateStatus(ctx, createOp, job); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (a *dataflowFlexTemplateJobAdapter) updateStatus(ctx context.Context, op directbase.Operation, job *pb.Job) error {
+	status := &krm.DataflowFlexTemplateJobStatus{
+		JobID:        job.GetId(),
+		CurrentState: direct.PtrTo(job.CurrentState.String()),
+		Type:         direct.PtrTo(job.Type.String()),
+	}
+
+	var readyCondition *v1alpha1.Condition
+
+	switch job.CurrentState {
+	case pb.JobState_JOB_STATE_RUNNING:
+		readyCondition = &v1alpha1.Condition{
+			Type:    v1alpha1.ReadyConditionType,
+			Status:  v1.ConditionTrue,
+			Reason:  k8s.UpToDate,
+			Message: "The resource is up to date",
 		}
-		switch latest.CurrentState {
-		case pb.JobState_JOB_STATE_RUNNING:
-			ready = true
-		default:
-			log.Info("unexpected job state waiting for job running", "state", latest.CurrentState)
+
+	case pb.JobState_JOB_STATE_FAILED:
+		readyCondition = &v1alpha1.Condition{
+			Type:    v1alpha1.ReadyConditionType,
+			Status:  v1.ConditionFalse,
+			Reason:  k8s.UpToDate,
```

Comment:
May I ask why this is UpToDate when the job is failed?

Reply:
Added a comment.  It's because (I think) we want to stop reconciliation at this point: we have reconciled the desired state and the outcome is a failed job.